agent/acs/handler/acs_handler.go

```diff
@@ -61,66 +61,30 @@ const (
 	sendCredentialsURLParameterName = "sendCredentials"
 )
 
-// StartSessionArguments is a struct representing all the things this handler
-// needs... This is really a hack to get by-name instead of positional
-// arguments since there are too many for positional to be wieldy
-type StartSessionArguments struct {
-	ContainerInstanceArn          string
-	CredentialProvider            *credentials.Credentials
-	Config                        *config.Config
-	DeregisterInstanceEventStream *eventstream.EventStream
-	TaskEngine                    engine.TaskEngine
-	ECSClient                     api.ECSClient
-	StateManager                  statemanager.StateManager
-	AcceptInvalidCert             bool
-	CredentialsManager            rolecredentials.Manager
+// Session encapsulates all arguments needed by the handler to connect to ACS
+// and to handle messages recieved by ACS. The Session.Start() method can be used
+// to start processing messages from ACS.
+type Session struct {
```

Comment:
Doing something like this makes it harder to test since you won't be able to stub out the `Session` struct with something else that fulfills the same interface.  Not a big deal, so if you feel strongly about it I'm not too concerned today.